_studio/mfx_lib/scheduler/src/mfx_scheduler_core.cpp

```diff
@@ -150,13 +149,10 @@ void mfxSchedulerCore::Close(void)
     }
 
     // delete task objects
-    for (n = 0; n < m_ppTaskLookUpTable.Size(); n += 1)
+    for (auto it : m_ppTaskLookUpTable)
     {
-        if (m_ppTaskLookUpTable[n])
-        {
-            delete m_ppTaskLookUpTable[n];
-            m_ppTaskLookUpTable[n] = NULL;
-        }
+        delete it;
+        it = nullptr;
```

Comment:
This will not affect the array, you need `auto& it` instead.

Reply:
I suggest to shrink vector to 0 after deletions instead of references usage.
Anyway, you need to do something with this code

Reply:
Following code is better from performance perspective:
for (auto ptr : m_ppTaskLookUpTable)
{
    delete ptr;
}
m_ppTaskLookUpTable.clear();